pkg/utils/sleeper_task.go

```diff
@@ -0,0 +1,129 @@
+package utils
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+// SleeperTask represents a task that waits in the background to process some work.
+type SleeperTask interface {
+	Stop() error
+	WakeUp()
+	WakeUpIfStarted()
+}
```

Comment:
In the spirit of 'accept interfaces, return structs', let's ditch this interface:
```suggestion
```
And move the comment to the struct, which we can then export as `SleeperTask`:
```go
// SleeperTask represents a task that waits in the background to process some work.
type SleeperTask struct {
```
And then `NewSleeperTask` can return `*SleeperTask`:
```go
func NewSleeperTask(worker Worker) *SleeperTask {
```

Reply:
But that means in field or variable declarations, like [here](https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/monitor/balance.go#L45) we will have a struct instead of an interface. Not sure if that's the best approach.

Reply:
Yes. That is not problematic and callers are still welcome to use the same exact interface for their own purposes if they want.

Reply:
It is idiomatic Go, and should our default choice until there is a specific reason to do otherwise.

Reply:
(it also makes testing way easier, and should allow us to avoid a cast in this case, and a particularly dangerous one that crosses package boundaries and uses an anonymous type)
https://github.com/smartcontractkit/chainlink/blob/71d314fe1b6aa1c35fc7c7e6d65525fee3d18bdd/core/chains/evm/monitor/balance_helpers_test.go#L4

Reply:
https://go.dev/blog/module-compatibility
> Where possible, it is better to avoid this class of problem entirely. When designing constructors, for example, prefer to return concrete types. Working with concrete types allows you to add methods in the future without breaking users, unlike interfaces. That property allows your module to be extended more easily in the future.